src/cmakeTaskProvider.ts

```diff
@@ -326,11 +335,16 @@ export class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.Outp
         if (generateLog) {
             telemetry.logEvent("task", {taskType: commandType, useCMakePresets: String(cmakeProject.useCMakePresets)});
         }
-        targets = await this.correctTargets(cmakeProject, commandType);
         const cmakeDriver: CMakeDriver | undefined = (await cmakeProject?.getCMakeDriverInstance()) || undefined;
         let cmakePath: string;
+        if(cmakeDriver)
```

Comment:
This implementation doesn't seem to be a good practice. The task with a specific target is created, but the code is altering the targets during the execution time, without any warning. If the targets needs to be changed, it should be before the task is being executed. 

Besides that, I don't think it is good idea to choose a defined task and then alter the targets. The `_findBuildTask` could work like this:
- fetch all of the tasks in tasks.json
- find the task that has the targets matching the targets chosen from the project outline
- if there is no such task, show a quickpick of the template tasks to choose from. 
         - alter the targets if the template task to match the project outline target.

Reply:
The question is what do you call 'template' task. I supose you mean these which are added by default by cmake Task Provider. 

E.g., I add task to tasks.json which does not have targets defined as it is tasks dedicated for all executable targets ( I add custom problem matcher I also seen cases when one was using tasks to get output in terminal which na n contrary to Output log support colours).

So in my case first condition to find tasks will never be met, second either as my taask is in tasks.json so wont appear on Quick pick list.

I agree that probably I could also eliminate in initial fetch all tasks which have target defined and it does not match the one requested.

Reply:
> The question is what do you call 'template' task. I supose you mean these which are added by default by cmake Task Provider.

correct.

> In my case I add task to tasks.json which does not have targets defined as it is task dedicated for all executable targets ( I add custom problem matcher I also seen cases when one was using tasks to get output in terminal which na n contrary to Output log support colours). I supose in your definitione it is not template task.

The user need to know what are the specifications of the task that is running. So it needs to be a custom task they defined in the tasks.json or the template task we provide. 

> Thus first condition to find tasks will never be met, second either as my taask is in tasks.json so wont appear on Quick pick list.

Would you please elaborate on this? 

> I agree that probably I could also eliminate in initial fetch all tasks which have target defined and it does not match the one requested.

Correct, quickpick should show the tasks with the same target, or template task for that target. And importantly, don't alter a selected task when running it.

Reply:
In template tasks, we only change the target, other characteristics stay the same.

Reply:
I reimplemented approach a little bit.

I list all cmake tasks.
Then I search for ones target matching to requested one. If more than one found - I search for one with default flag. Else I let use pick from list.
If no tasks with matching targets, then I do similar query among ones with no targets defined.

In tasks itself I use same "correctTargets" method which was already present. So all already implemented guards are in power. I added only a capability to query driver for "requested target".

By the way - I noticed that method runTestsTask and runConfigTasks also calls correctTargets but they ignore returned value. Does it make then sense to call them?